addons/stock_account/models/account_move.py

```diff
@@ -222,16 +261,118 @@ def _compute_account_id(self):
             if accounts['stock_input']:
                 line.account_id = accounts['stock_input']
 
+    def _create_in_invoice_svl(self):
+        svl_vals_list = []
+        for line in self:
+            line = line.with_company(line.company_id)
+            move = line.move_id.with_company(line.move_id.company_id)
+            po_line = line.purchase_line_id
+            uom = line.product_uom_id or line.product_id.uom_id
+
+            # Don't create value for more quantity than received
+            quantity = po_line.qty_received - (po_line.qty_invoiced - line.quantity)
```

Comment:
No unit conversions or price conversions are done here. I.e. PO line/invoice line is in Dozens, stock unit is Units -> wrong SVL entry created with a much higher price (price per dozen vs price per unit)